CMakeLists.txt

```
@@ -48,6 +48,84 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
+  if(DEFINED ENV{BUILDENV_BASEPATH})
+    set(BUILDENV_BASEPATH "$ENV{BUILDENV_BASEPATH}\\")
+  else(NOT DEFINED BUILDENV_BASEPATH)
+    set(BUILDENV_BASEPATH "${CMAKE_SOURCE_DIR}/buildenv/")
+  endif()
```

In other places we ignore environment variables if they are explicit passed to cmake via -D.
I think we should follow the same priority here and below with BUILDENV_URL

```suggestion
  if(NOT DEFINED BUILDENV_BASEPATH)
    if(DEFINED ENV{BUILDENV_BASEPATH})
      set(BUILDENV_BASEPATH "$ENV{BUILDENV_BASEPATH}\\")
    else()
      set(BUILDENV_BASEPATH "${CMAKE_SOURCE_DIR}/buildenv/")
    endif()
  endif()
```
We may even put this repeating pattern into a helper function.

---

It turns out that there is even a CMake bug in my mininal test. CMake forgets the variable when testing for the same variable on the environment.   

```
message(STATUS "-D ${BUILDENV_BASEPATH} ENV $ENV{BUILDENV_BASEPATH}")

if(NOT DEFINED BUILDENV_BASEPATH)
  message(status "1. BUILDENV_BASEPATH is not defined.")
endif()

if(DEFINED ENV{BUILDENV_BASEPATH})
  set(BUILDENV_BASEPATH "$ENV{BUILDENV_BASEPATH}\\")
else(NOT DEFINED BUILDENV_BASEPATH)
  message(status "2. BUILDENV_BASEPATH is not defined.")
  set(BUILDENV_BASEPATH "${CMAKE_SOURCE_DIR}/buildenv/")
endif()
  
message(FATAL_ERROR "3. ${BUILDENV_BASEPATH}")
```

```
$ cmake ../.. -DBUILDENV_BASEPATH="Hallo"
-- CMAKE_VERSION: 3.22.1
-- -D Hallo ENV 
status2. BUILDENV_BASEPATH is not defined.
CMake Error at CMakeLists.txt:64 (message):
  3.  /home/daniel/workspace/mixxx2/buildenv/
```

---

This comment is also pending